zerver/lib/digest.py

```diff
@@ -243,13 +243,17 @@ def get_stream_map(user_ids: List[int]) -> Dict[int, Set[int]]:
 
     stream_map = get_stream_map(user_ids)
 
+    long_term_idle_user_ids = [user.id for user in users if user.long_term_idle]
+
+    recently_modified_streams = get_modified_streams(long_term_idle_user_ids, cutoff_date)
+
     all_stream_ids = set()
 
     for user in users:
         stream_ids = stream_map[user.id]
 
         if user.long_term_idle:
-            stream_ids -= streams_recently_modified_for_user(user, cutoff_date)
+            stream_ids -= recently_modified_streams.get(user.id, set())
```

Comment:
Do we actually want the `long_term_idle` conditional here?  I think we may just want to exclude ALL streams where the user's subscription status changed, regardless of whether the user is long-term idle.

(I think the `long_term_idle` check was here only because of a perhaps mistaken belief that we were only looking at UserMessage rows, where (1) long_term_idle users don't have those and (2) UserMessage filtering would result in correct behavior around active users who were unsubscribed from a stream).  I think the simpler algorithm of just excluding everything where the user's subscription changed during the target window seems cleaner and more correct.

Reply:
Yeah, I was just preserving the behavior here, but I agree this would be a nice simplification.